src/ui/wxWidgets/PWSafeApp.cpp

```diff
@@ -496,12 +496,29 @@ bool PWSafeApp::OnInit()
       }
 #else
       wxMessageBox(_("Could not initialize help subsystem. Help will not be available."),
-      _("Password Safe: Error initializing help"), wxOK | wxICON_ERROR);
+                   _("Password Safe: Error initializing help"), wxOK | wxICON_ERROR);
 #endif
     }
   }
 
-  if (!cmd_closed && !cmd_silent && !cmd_minimized) {
+/*
+ * On macOS, Finder GUI file actions, such as double-clicking a file to open it, are sent
+ * to the app via calls to MacOpenFiles() and MacNewFile().  However, those messages
+ * are not processed until OnInit() returns.  This hack splits OnInit() into two parts,
+ * InitPart2() is called from those functions so that MacOpenFiles() has a chance
+ * to set the file name before the password entry dialog is displayed.
+ * All other information needed is stored in the PWSafeApp object.
+ * On Linux/Unix, just call the second part direct, so this is still like one big happy function.
+ */
+#ifdef __WXMAC__
+  return true;
+#else
+  return PWSafeApp::InitPart2();
+#endif //__WXMAC__
+}
+
+bool PWSafeApp::InitPart2() {
```

Comment:
"InitPart2" is a confusing name, I think, esp. since there's no InitPart1, and there's no indication if there should be InitPart3 or not.
"FinishInit" would be better, I think.
Also, since this always returns true, it could be void, and the calling logic could be changed accordingly.